datastore/postgres/migrations/matcher/14-v2-schema.sql

```diff
@@ -0,0 +1,585 @@
+	RETURNS BIGINT STRICT
+	LANGUAGE plpgsql
+	AS $$
+DECLARE
+	out BIGINT;
+BEGIN
+	INSERT INTO matcher_v2.package_name (name)
+		VALUES (n)
+	ON CONFLICT (name)
+		DO NOTHING
+	RETURNING
+		id INTO out;
+	RETURN out;
+END;
+$$;
+
+COMMENT ON FUNCTION package_name_id IS 'Return the ID for the provided name.';
+
+CREATE TABLE package (
+	id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	advisory BIGINT NOT NULL REFERENCES advisory (id) ON DELETE CASCADE,
+	name BIGINT NOT NULL REFERENCES package_name (id),
+	kind PackageKind,
+	arch Architecture[],
+	vulnerable_range VersionMultiRange NOT NULL CHECK (version_check (vulnerable_range)),
+	version_upstream TEXT[],
+	version_kind TEXT,
+	purl TEXT,
+	cpe TEXT,
+	UNIQUE (advisory, name, kind)
+);
+
+CREATE INDEX package_advisory_fk_idx ON package (advisory);
+
+CREATE INDEX package_package_name_fk_idx ON package (name);
+
+COMMENT ON TABLE package IS 'Package is the table containing software packages referenced in advisories. Packages are a special case of Attr.';
+
+COMMENT ON COLUMN package.advisory IS 'Foreign key to the advisory table.';
+
+COMMENT ON COLUMN package.name IS 'Foreign key to the package_name table.';
+
+COMMENT ON COLUMN package.kind IS 'The "kind" of this package.';
+
+COMMENT ON COLUMN package.arch IS 'Array of applicable architectures.';
+
+COMMENT ON COLUMN package.vulnerable_range IS 'The normalized version ranges of an affected package.';
+
+COMMENT ON COLUMN package.version_upstream IS 'The literal version strings as reported in the advisory.';
+
+COMMENT ON COLUMN package.version_kind IS 'The "kind" of the upstream version. Range operations are unlikely to make sense if this is not the expected kind.';
+
+COMMENT ON COLUMN package.purl IS 'The purl of this package. Version information should be omitted, as this is not describing a single package.';
+
+COMMENT ON COLUMN package.cpe IS 'The CPE for this package.';
+
+COMMENT ON INDEX package_advisory_fk_idx IS 'Index on foreign key for less costly deletes & joins.';
+
+COMMENT ON INDEX package_package_name_fk_idx IS 'Index on foreign key for less costly deletes & joins.';
+-- }}}
+-- {{{ Attr table
+CREATE TABLE mediatype (
+	id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	mediatype TEXT UNIQUE NOT NULL
+);
+
+COMMENT ON TABLE mediatype IS 'Mediatype is the table for interning mediatypes.';
+
+COMMENT ON COLUMN mediatype.mediatype IS 'Media type. Should have an "application" type, although this is not currently enforced.';
+
+CREATE TABLE attr (
+	id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	mediatype BIGINT NOT NULL REFERENCES mediatype (id) ON DELETE CASCADE,
+	data JSONB NOT NULL
+);
+
+CREATE INDEX attr_mediatype_fk_idx ON attr (mediatype);
+
+CREATE INDEX attr_data_gidx ON attr USING GIN (data jsonb_path_ops);
+
+CREATE INDEX attr_data_hidx ON attr USING HASH (data);
+
+ALTER TABLE attr
+	ADD UNIQUE (mediatype, data);
+
+COMMENT ON TABLE attr IS $$
+Attrs are generic features of advisories and references.
```

Comment:
I don't see a table to connect `attr`s with `reference`s

Reply:
Right, `attr` and `reference` are associated with an `advisory`.